plugins/pluginmanagers/lazy.nix

```
@@ -1,222 +1,264 @@
+
+                optional = defaultNullOpts.mkBool false ''
+                  Optional specs are only included in the final configuration
+                  if the corresponding plugin is also specified as a required
+                  (non-optional) plugin elsewhere. This feature is
+                  particularly helpful for Neovim distributions, allowing
+                  them to pre-configure settings for plugins that users may
+                  or may not have installed.
+                '';
+
+                opts = mkNullOrOption (maybeRaw (attrsOf anything)) ''
+                  The opts value can be one of the following:
+
+                  - A table: This table will be merged with any existing
+                  configuration settings from parent specifications.
+                  - A function that returns a table: The returned table will
+                  completely replace any existing configuration settings from
+                  parent specifications.
+                  - A function that modifies a table: This function will
+                  receive the existing configuration table as an argument and
+                  can modify it directly.
+
+                  In all cases, the resulting configuration table will be
+                  passed to the Plugin.config() function. Setting the opts
+                  value automatically implies that Plugin.config() will be
+                  called. (See: https://lazy.folke.io/spec#spec-setup)
+                '';
+              };
+
+              config.name = lib.mkIf (cfg.pkg != null) (lib.mkDefault "${lib.getName cfg.pkg}");
+              config.dir = lib.mkIf (cfg.pkg != null) (lib.mkDefault "${cfg.pkg}");
+            }
+          )
+        );
+
+      lazyPluginsListType = types.listOf lazyPluginType;
+    in
+    {
+      gitPackage = lib.mkPackageOption pkgs "git" { nullable = true; };
+      luarocksPackage = lib.mkPackageOption pkgs "luarocks" { nullable = true; };
+
+      plugins = lib.mkOption {
+        type = lazyPluginsListType;
+        default = [ ];
+        description = "List of plugins";
+      };
+    };
+
+  extraConfig =
+    cfg:
+    let
+      # The `pkg` property isn't a part of the `lazy.nvim` plugin spec, while
+      # it shouldn't do any harm but it does take up unnecessary space in the
+      # init.lua file. Since we're done using it we will strip it from the
+      # final list of specs.
+      removePkgAttrFromPlugin =
+        plugin:
+        builtins.removeAttrs plugin [ "pkg" ]
+        // lib.optionalAttrs (((plugin.dependencies or null) != null) && lib.isList plugin.dependencies) {
+          dependencies = map removePkgAttrFromPlugin plugin.dependencies;
         };
+
+      removePkgAttrFromPlugins = plugins: map removePkgAttrFromPlugin plugins;
+    in
+    {
+      extraPackages = [
+        cfg.gitPackage
+        cfg.luarocksPackage
+      ];
+      plugins.lazy.settings.spec = removePkgAttrFromPlugins cfg.plugins;
```

```suggestion

      plugins.lazy.settings.spec = map toPluginSpec cfg.plugins;
```

Let's inline `removePkgAttrFromPlugins` here and rename `removePkgAttrFromPlugin` to `toPluginSpec`.